### ISSUE TITLE
[Bug]: Nav - selected category is collapsed on mount, selectedCategoryValue doesn't control it

### ISSUE THREAD

**Issue by slimklim** [microsoft/fluentui#32417]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: macOS 14.6.1
    CPU: (11) arm64 Apple M3 Pro
    Memory: 417.91 MB / 36.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 128.0.6613.114
    Edge: 128.0.2739.42
    Firefox: 129.0.2
    Safari: 17.6
```

### Are you reporting an Accessibility issue?

no

### Reproduction

https://stackblitz.com/edit/yrf6gk?file=src%2Fexample.tsx

## Actual Behavior
selected category is collapsed when the component is rendered for the first time and selectedCategoryValue is set up

## Expected Behavior
selected category is expanded when the component is rendered for the first time and selectedCategoryValue is set up

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Hotell** [comment]
@slimklim just to verify 

Current:

<img width="1210" alt="image" src="https://github.com/user-attachments/assets/dc5e082a-f3ea-4c17-872b-2f553fa40e44">


Expected:

<img width="323" alt="image" src="https://github.com/user-attachments/assets/6e93d56f-f975-4a2a-b9af-ac113d9d9329">

**Comment by slimklim** [comment]
> @slimklim just to verify
> 
> Current:
> 
> <img alt="image" width="1210" src="https://private-user-images.githubusercontent.com/1223799/362794765-dc5e082a-f3ea-4c17-872b-2f553fa40e44.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzNzYxNTQsIm5iZiI6MTcyNTM3NTg1NCwicGF0aCI6Ii8xMjIzNzk5LzM2Mjc5NDc2NS1kYzVlMDgyYS1mM2VhLTRjMTctODcyYi0yZjU1M2ZhNDBlNDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDNUMTUwNDE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGYzODI0MjIyYjg5NzI1ZDJkNDg4NWU1OTY5MjM2ZGY3MjRhNWM0MTIzY2ViNzI3ZDUxNTI0NzJjNzg4YjczNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.o4kyPqcoIiLXnN81UGkLfhk7EfJ1EE6-3x-ez4Mn5VE">
> Expected:
> 
> <img alt="image" width="323" src="https://private-user-images.githubusercontent.com/1223799/362797937-6e93d56f-f975-4a2a-b9af-ac113d9d9329.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzNzYxNTQsIm5iZiI6MTcyNTM3NTg1NCwicGF0aCI6Ii8xMjIzNzk5LzM2Mjc5NzkzNy02ZTkzZDU2Zi1mOTc1LTRhMmEtYjlhZi1hYzExM2Q5ZDkzMjkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDNUMTUwNDE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2IwNzYwZTdiMTRmMjlmYjA4ZjM0NGU0YmJiNzI3YmJjZjE3YjRiMmQ2NTEzMTU0NjExZDc3MmUzYzkzYWIyNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dLhPlBnxwWQJwG5mrdtfZxAVLkjZd7X-RZ1Raf_cp3w">

exactly

**Comment by mltejera** [comment]
Thanks again for filing @slimklim - this should be fixed with #32489
